### ISSUE TITLE
k8s ingress error after upgrade

### ISSUE THREAD

**Issue by michaelwasser** [DNXLabs/terraform-aws-eks-lb-controller#26]
### Contact Details (optional)

michael.wasserstrum@niceactimize.com

### Description. What happened?

After upgrade of EKS and aws-load-balancer-controller, a k8s ingress is not created as expected.

### Steps to reproduce

Before upgrade:
EKS 1.21
app.kubernetes.io/version=v2.3.0
helm.sh/chart=aws-load-balancer-controller-1.3.2

After upgrade:
EKS 1.22
app.kubernetes.io/version=v2.4.3
helm.sh/chart=aws-load-balancer-controller-1.4.4

### Expected behavior

k8s ingress on cm-michael1 namespace is created successfully

### Actual behavior

k8s ingress on cm-michael1 namespace is created with an error:

Failed build model due to ingress: cm-michael1/michael1-actone: secrets "michael1-cxone-creds" is forbidden: User "system:serviceaccount:kube-system:aws-alb-ingress-controller" cannot get resource "secrets" in API group "" in the namespace "cm-michael1"

### Occurrence

Frequently

### Relevant log output

```shell
kubectl get ingress michael1-actone -n cm-michael1

Name:             michael1-actone
Namespace:        cm-michael1
Address:
Default backend:  default-http-backend:80 (<error: endpoints "default-http-backend" not found>)
Rules:
  Host                                     Path  Backends
  ----                                     ----  --------
  cmaas.dev-cm.niceactimize-xsightdev.com
                                           /actone/michael1/*   michael1-actone:use-annotation ()
Annotations:                               alb.ingress.kubernetes.io/actions.michael1-actone:
                                             {"Type":"forward","ForwardConfig":{"TargetGroups":[{"ServiceName":"michael1-actone","ServicePort":"8443"}],"TargetGroupStickinessConfig":{...
                                           alb.ingress.kubernetes.io/auth-idp-oidc:
                                             {"issuer":"https://cxone.test.niceincontact.com","authorizationEndpoint":"https://cxone.test.niceincontact.com/auth/authorize","tokenEndpo...
                                           alb.ingress.kubernetes.io/auth-on-unauthenticated-request: authenticate
                                           alb.ingress.kubernetes.io/auth-scope: openid
                                           alb.ingress.kubernetes.io/auth-type: oidc
                                           alb.ingress.kubernetes.io/backend-protocol: HTTPS
                                           alb.ingress.kubernetes.io/conditions.michael1-actone: [{"field":"path-pattern","pathPatternConfig":{"values":["/actone/michael1"]}}]
                                           alb.ingress.kubernetes.io/group.name: public-alb-dev-cm
                                           alb.ingress.kubernetes.io/healthcheck-interval-seconds: 10
                                           alb.ingress.kubernetes.io/healthcheck-path: /actone/michael1/api/public/v1/heartbeat
                                           alb.ingress.kubernetes.io/healthcheck-port: traffic-port
                                           alb.ingress.kubernetes.io/healthcheck-protocol: HTTPS
                                           alb.ingress.kubernetes.io/healthcheck-timeout-seconds: 5
                                           alb.ingress.kubernetes.io/healthy-threshold-count: 2
                                           alb.ingress.kubernetes.io/listen-ports: [{"HTTPS":443}]
                                           alb.ingress.kubernetes.io/success-codes: 200
                                           alb.ingress.kubernetes.io/target-group-attributes: stickiness.enabled=true,stickiness.lb_cookie.duration_seconds=3600
                                           alb.ingress.kubernetes.io/target-type: ip
                                           alb.ingress.kubernetes.io/unhealthy-threshold-count: 3
Events:
  Type     Reason            Age                 From     Message
  ----     ------            ----                ----     -------
  Warning  FailedBuildModel  16m (x19 over 38m)  ingress  Failed build model due to ingress: cm-michael1/michael1-actone: secrets "michael1-cxone-creds" is forbidden: User "system:serviceaccount:kube-system:aws-alb-ingress-controller" cannot get resource "secrets" in API group "" in the namespace "cm-michael1"

------------------------------------------------------------------------------

kubectl describe deployment.apps/aws-load-balancer-controller -n kube-system

Name:                   aws-load-balancer-controller
Namespace:              kube-system
CreationTimestamp:      Mon, 22 Aug 2022 07:58:01 +0000
Labels:                 app.kubernetes.io/instance=aws-load-balancer-controller
                        app.kubernetes.io/managed-by=Helm
                        app.kubernetes.io/name=aws-load-balancer-controller
                        app.kubernetes.io/version=v2.4.3
                        helm.sh/chart=aws-load-balancer-controller-1.4.4
Annotations:            deployment.kubernetes.io/revision: 1
                        meta.helm.sh/release-name: aws-load-balancer-controller
                        meta.helm.sh/release-namespace: kube-system
Selector:               app.kubernetes.io/instance=aws-load-balancer-controller,app.kubernetes.io/name=aws-load-balancer-controller
Replicas:               2 desired | 2 updated | 2 total | 2 available | 0 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  25% max unavailable, 25% max surge
Pod Template:
  Labels:           app.kubernetes.io/instance=aws-load-balancer-controller
                    app.kubernetes.io/name=aws-load-balancer-controller
  Annotations:      prometheus.io/port: 8080
                    prometheus.io/scrape: true
  Service Account:  aws-alb-ingress-controller
  Containers:
   aws-load-balancer-controller:
    Image:       602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon/aws-load-balancer-controller:v2.4.3
    Ports:       9443/TCP, 8080/TCP
    Host Ports:  0/TCP, 0/TCP
    Command:
      /controller
    Args:
      --cluster-name=cmaas-dev-cm-eks
      --ingress-class=alb
    Liveness:     http-get http://:61779/healthz delay=30s timeout=10s period=10s #success=1 #failure=2
    Environment:  <none>
    Mounts:
      /tmp/k8s-webhook-server/serving-certs from cert (ro)
  Volumes:
   cert:
    Type:               Secret (a volume populated by a Secret)
    SecretName:         aws-load-balancer-tls
    Optional:           false
  Priority Class Name:  system-cluster-critical
Conditions:
  Type           Status  Reason
  ----           ------  ------
  Available      True    MinimumReplicasAvailable
  Progressing    True    NewReplicaSetAvailable
OldReplicaSets:  <none>
NewReplicaSet:   aws-load-balancer-controller-66b449b74 (2/2 replicas created)
Events:
  Type    Reason             Age   From                   Message
  ----    ------             ----  ----                   -------
  Normal  ScalingReplicaSet  59m   deployment-controller  Scaled up replica set aws-load-balancer-controller-66b449b74 to 2
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hardbyte** [comment]
I've run into this as well. Appears to be two possible workarounds:

1) Add a Role and RoleBinding that explicitly grants access to the specific required secrets to the service account. E.g. https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/examples/secrets_access/
2) explicitly set `clusterSecretsPermissions.allowAllSecrets`

**Comment by jeremiasroma** [comment]
Fix applied to the [release 0.7.0](https://github.com/DNXLabs/terraform-aws-eks-lb-controller/releases/tag/0.7.0)
